### PR TITLE
(maint) Add steps for opening a PR to contributing

### PR DESCRIPTION
**Pull request by lucywyman**
This adds more detailed steps to our workflow for having contributors open PRs against Bolt. It includes forking, cloning, and branching methods that contributors should follow.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -32,6 +32,13 @@ Pull requests are also welcome on GitHub: https://github.com/puppetlabs/bolt
 As with other open-source projects managed by Puppet, you must digitally sign the Contributor
 License Agreement before we can accept your pull request: https://cla.puppet.com
 
+To submit a PR:
+1. Fork the Bolt project (button in the top-right, next to 'star' and 'watch')
+2. Clone your fork of Bolt
+3. Make a new branch off of master - `git checkout -b mybranchname`
```
Comment (donoghuc) [inline CONTRIBUTING.md]: Is adding the upstream repo too advanced for this? This works fine for the case where someone wants to make a quick one-off change, but if they come back to it later their fork will have gotten out of date.

Comment (lucywyman) [reply CONTRIBUTING.md]: Great point! Updated

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -32,6 +32,25 @@ Pull requests are also welcome on GitHub: https://github.com/puppetlabs/bolt
...
+2. Clone your fork of Bolt
+3. Add the puppetlabs repo as an upstream - `git remote add puppet git@github.com:puppetlabs/bolt`
+4. Make a new branch off of master - `git checkout -b mybranchname`
+5. Push your changes to your branch on your fork - `git push origin mybranchname`
+6. Open a PR against master at https://github.com/puppetlabs/bolt
+
+**If it's not your first PR**
+1. Update from upstream:
+   ```
+   git checkout master
```
Comment (donoghuc) [inline CONTRIBUTING.md]: Could avoid rebase by
```
git fetch puppet
git checkout puppet/master
git checkout -b mynewbranch
```

**Comment by binford2k** [comment]
Is it worth updating the [global CONTRIBUTING.md](https://github.com/puppetlabs/.github/blob/master/CONTRIBUTING.md) doc with standard info?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -32,6 +32,24 @@ Pull requests are also welcome on GitHub: https://github.com/puppetlabs/bolt
 As with other open-source projects managed by Puppet, you must digitally sign the Contributor
 License Agreement before we can accept your pull request: https://cla.puppet.com
 
+**If this is your first time submitting a PR**:
+1. Fork the Bolt project (button in the top-right, next to 'star' and 'watch')
+2. Clone your fork of Bolt
+3. Add the puppetlabs repo as an upstream - `git remote add puppet git@github.com:puppetlabs/bolt`
+4. Make a new branch off of master - `git checkout -b mybranchname`
```
Comment (adreyer) [inline CONTRIBUTING.md]: `git fetch remote && git checkout remote/master && git checkout -b mybranchname`

4.5. Add commits to your branch?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -32,6 +32,24 @@ Pull requests are also welcome on GitHub: https://github.com/puppetlabs/bolt
...
+5. Push your changes to your branch on your fork - `git push origin mybranchname`
+6. Open a PR against master at https://github.com/puppetlabs/bolt
+
+**If it's not your first PR**
+1. Update from upstream:
+   ```
+   git fetch puppet
```
Comment (adreyer) [inline CONTRIBUTING.md]: this remote name doesn't align with the previous use of "remote"

**Comment by lucywyman** [comment]
@binford2k That'd be awesome! I glanced through the Puppet one and didn't see anything useful - where would a global contributing doc live?

**Comment by binford2k** [comment]
@lucywyman if a repo in the `puppetlabs` namespace doesn't have a `CONTRIBUTING.md` right now, then it will be sourced from https://github.com/puppetlabs/.github/blob/master/CONTRIBUTING.md

That's brand new, they just announced that feature this year. So we don't yet have a best practices around using it.

**Pull request merged**